modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -1031,6 +1031,20 @@ private String _getDockerImageId(Project project) {
 		return project.getName() + ":latest";
 	}
 
+	private String _getFolderPath(
+		String path, Project project, String defaultValue) {
+
+		String virtualInstanceId = (String)GradleUtil.getProperty(
+			project.getRootProject(), "liferay.virtual.instance.id",
+			defaultValue);
+
+		if ((virtualInstanceId != null) && !virtualInstanceId.isEmpty()) {
+			return path + "/" + virtualInstanceId;
+		}
+
+		return path;
+	}
```

Comment:
I might split this up into a couple of methods - one dedicated to retrieving the `virtualInstanceId`, and one for appending the suffix to the path if the suffix is real (like in the other pull request).

The use of `defaultValue` in this one could be misleading - default value for what case? However, if it was used in the case of a `_getVirtualInstanceId` method, it would be pretty clear what is meant by it, since it would truly function as a default value the way we see in most other cases.

Reply:
For this I would add these methods: 
`_getVirtualInstanceId(Project project, String defaultValue)`
`_getFolderPath(String path, String virtualInstanceId)`

In the `_getFolderPath` method I'd check the virtualInstanceId (`Validator.isNotNull(virtualInstanceId)`) and return the path based on that (with or without the suffix). 

Lastly I would call the `_getFolderPath` everywhere similarly to what I've done before but using the `_getVirtualInstanceId` method, for example:
`_getFolderPath("client-extensions", _getVirtualInstanceId(project, ""))`

What do you think?

Reply:
Yes that seems good 👍 

I was just playing around with `Paths.get`, and discovered that if you give it a blank string as a path segment, it will collapse the path. We could take advantage of this when constructing these strings: 

```java
Assert.assertEquals("foo/baz", Paths.get("foo", "", "baz").toString());
```